pkg/virtctl/portforward/portforward.go

```diff
@@ -0,0 +1,220 @@
+	if err != nil {
+		return err
+	}
+
+	if templates.KindIsVMI(kind) {
+		o.resource = client.VirtualMachineInstance(namespace)
+	} else if templates.KindIsVMI(kind) {
+		o.resource = client.VirtualMachine(namespace)
+	} else {
+		return errors.New("unsupported resource kind " + kind)
+	}
+
+	return nil
+}
+
+func (o *PortForward) startStdoutStream(namespace, name string, port forwardedPort) error {
+	streamer, err := o.resource.PortForward(name, port.remote, port.protocol)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(os.Stderr, "forwarding to %s/%s:%d\n", namespace, name, port.remote)
+	if err := streamer.Stream(kubecli.StreamOptions{
+		In:  os.Stdin,
+		Out: os.Stdout,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *PortForward) startPortForwards(namespace, name string, ports []forwardedPort) error {
+	for _, port := range ports {
+		forwarder := portForwarder{
+			namespace: namespace,
+			name:      name,
+			resource:  o.resource,
+		}
+		if err := forwarder.startForwarding(o.address, port); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func usage() string {
+	return `Forward local ports to a virtualmachine or virtualmachineinstance.
+	
+The target argument supports the syntax kind/name.namespace with kind/ and .namespace as optional fields.
+Kind accepts any of vmi (default), vmis, vm, vms, virtualmachineinstance, virtualmachine, virtualmachineinstances, virtualmachines.
+
+The port argument supports the syntax protocol/localPort:targetPort with protocol/ and :targetPort as optional fields.
+Protocol supports TCP (default) and UDP.
+
+Portforwards get established over the Kubernetes control-plane using websocket streams.
+Usage can be restricted by the cluster administrator through the /portforward subresource.
+`
+}
+
+func examples() string {
+	return `  # Forward the local port 8080 to the vmi port:
+  {{ProgramName}} port-forward vmi/testvmi 8080
+
+  # Forward the local port 8080 to the vmi port 9090:
+  {{ProgramName}} port-forward vmi/testvmi 8080:9090
+
+  # Forward the local port 8080 to the vmi port 9090 as a UDP connection:
+  {{ProgramName}} port-forward vmi/testvmi.mynamespace udp/8080:9090
+
+  # Forward the local port 8080 to the vm port
+  {{ProgramName}} port-forward vm/testvm 8080
+
+  # Forward the local port 8080 to the vm port in mynamespace
+  {{ProgramName}} port-forward vm/testvm.mynamespace 8080
+
+  # Note: {{ProgramName}} port-forward sends all traffic over the Kubernetes API Server. 
+  # This means any traffic will add additional pressure to the control plane.
+  # For continous traffic intensive connections, consider using a dedicated Kubernetes Service.
+
+  # Open an SSH connection using PortForward and ProxyCommand:
+  ssh -o 'ProxyCommand={{ProgramName}} port-forward --stdio=true testvmi 22' user@testvmi.mynamespace
+
+  # Use as SCP ProxyCommand:
+  scp -o 'ProxyCommand={{ProgramName}} port-forward --stdio=true testvmi 22' local.file user@testvmi.mynamespace`
```

Comment:
The commands will work but are probably slightly incorrect in the sense that `mynamespace` will not be used and it will just use the current namespace, right?

Reply:
True, I added the namespace to the port-forward command to make it more correct.